core/trino-main/src/main/java/io/trino/execution/scheduler/HashDistributionSplitAssigner.java

```diff
@@ -233,4 +347,29 @@ public int getId()
             return id.getAsInt();
         }
     }
+
+    private static boolean isWriteFragment(PlanFragment fragment)
+    {
+        PlanVisitor<Boolean, Void> visitor = new PlanVisitor<>()
+        {
+            @Override
+            protected Boolean visitPlan(PlanNode node, Void context)
+            {
+                for (PlanNode child : node.getSources()) {
```

Comment:
This check is used to prevent partition merging (to avoid running into number of partitions limit for TableWrite). As long as there's a single TableWriter node (no matter where) it should be safe to assume that a fragment is a TableWrite fragment.

Reply:
Yeah - I just meant that any node which has a TableWriter as a child has just a single node. Or not necessarily?

Anyway - no changes requestes - just a curious question.

Reply:
It should be true within a single fragment. It is not necessarily true overall due to `PushTableWriteThroughUnion`.